src/ipc-handlers.ts

```diff
@@ -542,6 +542,14 @@ export async function saveOnboarding(
 	} );
 }
 
+export async function saveUserLocale( _event: IpcMainInvokeEvent, userLocale: string ) {
+	const userData = await loadUserData();
+	await saveUserData( {
+		...userData,
+		userLocale,
+	} );
```

Comment:
I think this approach will be a bit limiting until userData could be saved and loaded across both the main and renderer threads.